### PR TITLE
Unrevivable trait

### PR DESCRIPTION
**Pull request by Scribbles0**
## About the PR

Adds the unrevivable trait, which makes you unable to be revived by defibrillator.

## Why / Balance

Allows for permadeath which can affect the flow of a round in interesting ways and/or add unique roleplay scenarios.

## Technical details

Replaces/removes any remnants of the unclonable trait.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/space-wizards/space-station-14/assets/91828755/ea4c616c-c459-4a9b-82e6-f3b4fb94a12a

**Changelog**

:cl:
- add: Added a new trait, the Unrevivable trait.

### CODE REVIEW

**Comment by UbaserB** [comment]
Based!

**Comment by UbaserB** [comment]
Hold up, this conflicts with #24216

File: Content.Server/Medical/DefibrillatorSystem.cs [hunk 1]
```diff
@@ -221,6 +222,11 @@ public void Zap(EntityUid uid, EntityUid target, EntityUid user, DefibrillatorCo
             _chatManager.TrySendInGameICMessage(uid, Loc.GetString("defibrillator-rotten"),
                 InGameICChatType.Speak, true);
         }
+        else if (TryComp<UnrevivableComponent>(target, out var comp))
```
Comment (EmoGarbage404) [inline Content.Server/Medical/DefibrillatorSystem.cs:225]: HasComp

**Comment by UbaserB** [comment]
Based

**Pull request merged**